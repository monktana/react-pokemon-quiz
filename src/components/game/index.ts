export * from './Game';
export * from './useGuess';
