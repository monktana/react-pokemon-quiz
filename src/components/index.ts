export * from './error';
export * from './game';
export * from './gameover';
export * from './language';
export * from './loading';
export * from './menu';
export * from './navbar';
export * from './pokemon';
export * from './question';
export * from './score';
