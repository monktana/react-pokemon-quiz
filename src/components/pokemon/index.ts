export * from './Pokemon';
export * from './PokemonName';
export * from './PokemonSprite';
export * from './PokemonTags';
export * from './TypeTag';
export * from './util';
export * from './pokemon-context';
