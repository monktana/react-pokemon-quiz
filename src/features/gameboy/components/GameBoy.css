.gameboy-container {
  /* Original: 84,6 (w) X  82 (h) mm */
  width: calc(500px * 1.16); /* 580px */
  height: calc(1000px * 1.095); /* 1095px */

  position: relative;

  background-color: transparent;
  border: 3px solid black;
}

.gameboy-container .top-half {
  height: calc((1000px * 1.095) / 2 - 40px); /* 508px */

  position: relative;

  border: 3px solid pink;

  border-top-left-radius: 25px;
  border-top-right-radius: 25px;
  border-bottom-left-radius: 15px;
  border-bottom-right-radius: 15px;
}

.gameboy-container .top-half .screen-container {
  margin: auto;

  position: absolute;
  top: 55px;
  left: 40px;
  bottom: 65px;
  right: 30px;

  border: 3px solid aqua;

  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
}

.gameboy-container .top-half .screen-container .screen {
  margin: auto;

  position: absolute;
  top: 25px;
  left: 25px;
  bottom: 55px;
  right: 25px;
  
  border: 3px solid blue;

  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
}

.gameboy-container .bottom-half {
  height: calc((1000px * 1.095) / 2 + 40px); /* 548px */

  position: relative;

  border: 3px solid orange;

  border-top-left-radius: 25px;
  border-top-right-radius: 25px;
  border-bottom-left-radius: 50px;
  border-bottom-right-radius: 50px;
}

.gameboy-container .bottom-half .controlls-container {
  margin: auto;

  position: absolute;
  top: 75px;
  left: 0;
  bottom: 0;
  right: 0;

  border: 3px solid aqua;

  border-bottom-left-radius: 50px;
  border-bottom-right-radius: 50px;
}

.gameboy-container .bottom-half .controlls-container .cross {
  width: 200px;
  height: 200px;

  margin: auto;

  position: absolute;
  top: 85px;
  left: 40px;
  
  border: 3px solid blue;
  border-radius: 100%;
}

.gameboy-container .bottom-half .controlls-container .buttons-a-b {
  width: 200px;
  height: 110px;

  margin: auto;

  position: absolute;
  top: 125px;
  right: 40px;
  
  transform: rotate(-15deg);

  border: 3px solid blue;
  border-radius: 35%;
}

.gameboy-container .bottom-half .controlls-container .buttons-start-select {
  width: 175px;
  height: 85px;

  margin: auto;

  position: absolute;
  left: 0;
  right: 0;
  bottom: 20px;

  border: 3px solid blue;
  border-radius: 35%;
}