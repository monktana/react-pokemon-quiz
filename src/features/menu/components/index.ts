export * from './menu';
export * from './navbar';
