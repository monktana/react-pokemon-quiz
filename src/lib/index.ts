export * from './ky';
export * from './react-query';
