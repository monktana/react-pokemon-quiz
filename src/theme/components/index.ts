export * from './Button';
export * from './Pokemon';
export * from './Skeleton';
