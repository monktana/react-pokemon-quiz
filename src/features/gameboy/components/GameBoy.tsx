import { PropsWithChildren } from 'react'
import './GameBoy.css'

type GameBoyProps = PropsWithChildren<{
  placeholder?: string
}>

export function GameBoy({children}: GameBoyProps) {
  return (
    <div className="gameboy-container">
      <div className="top-half">
        <div className="screen-dot"></div>
        <div className="screen-dot"></div>
        <div className="screen-dot"></div>
        <div className="screen-dot"></div>
        <div className="screen-container">
          <div className="screen-headline">
            <span>GAMEBOY ADVANCE SP</span>
          </div>
          <div className="screen">
            {children}
          </div>
        </div>
      </div>
      <div className="bottom-half">
        <div className="controlls-container">
          <div className="brightness"></div>
          <div className="cross"></div>
          <div className="buttons-a-b"></div>
          <div className="buttons-start-select"></div>
        </div>
      </div>
    </div>
  )
}