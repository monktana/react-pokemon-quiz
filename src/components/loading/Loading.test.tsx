import React from 'react';
import { screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Loading } from '@/components';
import { render } from '@/lib';

describe('Loading', () => {
  it('renders the Loading component ', () => {
    render(<Loading />);
    expect(screen.getByTestId('loading-container')).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
  });
});
