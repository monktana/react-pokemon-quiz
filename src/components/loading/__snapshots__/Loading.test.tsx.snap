// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading > matches the snapshot 1`] = `
<DocumentFragment>
  <script
    id="chakra-script"
  >
    !(function(){try{var a=function(c){var v="(prefers-color-scheme: dark)",h=window.matchMedia(v).matches?"dark":"light",r=c==="system"?h:c,o=document.documentElement,s=document.body,l="chakra-ui-light",d="chakra-ui-dark",i=r==="dark";return s.classList.add(i?d:l),s.classList.remove(i?l:d),o.style.colorScheme=r,o.dataset.theme=r,r},n=a,m="system",e="chakra-ui-color-mode",t=localStorage.getItem(e);t?a(t):localStorage.setItem(e,a(m))}catch(a){}})();
  </script>
  <div
    class="css-h74a88"
    data-sentry-component="Loading"
    data-sentry-element="Center"
    data-sentry-source-file="Loading.tsx"
  >
    <div
      class="chakra-stack css-bfgy4l"
      data-sentry-element="VStack"
      data-sentry-source-file="Loading.tsx"
      data-testid="loading-container"
    >
      <div
        class="chakra-skeleton css-1eh6o9k"
        data-sentry-element="Skeleton"
        data-sentry-source-file="Loading.tsx"
      />
      <div
        class="chakra-skeleton css-11jpbuw"
        data-sentry-element="Skeleton"
        data-sentry-source-file="Loading.tsx"
      />
      <div
        class="chakra-skeleton css-11jpbuw"
        data-sentry-element="Skeleton"
        data-sentry-source-file="Loading.tsx"
      />
      <div
        class="chakra-skeleton css-1eh6o9k"
        data-sentry-element="Skeleton"
        data-sentry-source-file="Loading.tsx"
      />
      <div
        class="css-e3gl6g"
        data-sentry-element="Grid"
        data-sentry-source-file="Loading.tsx"
      >
        <div
          class="chakra-skeleton css-10yr3u1"
          data-sentry-element="Skeleton"
          data-sentry-source-file="Loading.tsx"
        />
        <div
          class="chakra-skeleton css-10yr3u1"
          data-sentry-element="Skeleton"
          data-sentry-source-file="Loading.tsx"
        />
        <div
          class="chakra-skeleton css-10yr3u1"
          data-sentry-element="Skeleton"
          data-sentry-source-file="Loading.tsx"
        />
        <div
          class="chakra-skeleton css-10yr3u1"
          data-sentry-element="Skeleton"
          data-sentry-source-file="Loading.tsx"
        />
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
